src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -31,23 +31,55 @@ class ConfigurationHelper implements Serializable {
         if(!this.name) throw new IllegalArgumentException('Step has no public name property!')
     }
 
+    /*
+     * By default this methods does nothing. With this method we are able to ensure that we do not call the
+     * deprecated methods. Might be usefull during local development.
+     */
+    private static handleDeprecation(script, String methodName) {
+        if(script != null) {
+            def msg = "ConfigurationHelper.${methodName} was called with a script reference." +
+                'This method is deprecated. Use the same method without the script reference'
+            if(Boolean.getBoolean('com.sap.piper.failOnScriptReferenceInConfigurationHelper'))
+                throw new RuntimeException(msg)
+            if(Boolean.getBoolean('com.sap.piper.emitWarningOnScriptReferenceInConfigurationHelper') &&
+                script instanceof Script) script.echo("[WARNING] ${msg}")
+        }
+    }
+
+
     ConfigurationHelper collectValidationFailures() {
         validationResults = validationResults ?: [:]
         return this
     }
 
+    ConfigurationHelper mixinGeneralConfig(Set filter = null, Map compatibleParameters = [:]){
+        mixinGeneralConfig(null, filter, compatibleParameters)
+    }
+    @Deprecated
+    /** Use  mixinGeneralConfig without commonPipelineEnvironment*/
     ConfigurationHelper mixinGeneralConfig(commonPipelineEnvironment, Set filter = null, Map compatibleParameters = [:]){
-        Map generalConfiguration = ConfigurationLoader.generalConfiguration([commonPipelineEnvironment: commonPipelineEnvironment])
+        handleDeprecation(commonPipelineEnvironment, 'mixinGeneralConfig')
+        Map generalConfiguration = ConfigurationLoader.generalConfiguration()
         return mixin(generalConfiguration, filter, compatibleParameters)
     }
 
+    ConfigurationHelper mixinStageConfig(stageName, Set filter = null, Map compatibleParameters = [:]){
+        mixinStageConfig(null, stageName, filter, compatibleParameters)
+    }
+    @Deprecated
     ConfigurationHelper mixinStageConfig(commonPipelineEnvironment, stageName, Set filter = null, Map compatibleParameters = [:]){
-        Map stageConfiguration = ConfigurationLoader.stageConfiguration([commonPipelineEnvironment: commonPipelineEnvironment], stageName)
+        handleDeprecation(commonPipelineEnvironment, 'mixinStageConfig')
```

Comment:
The handleDeprecation and also the error message states that you should not call the methods with a `script` reference.  However, it is anyway the `commonPipelineEnvironment`.